classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,378 @@
+<?php
+
+/**
+ * Implements the admin view of the actions.
+ */
+class ActionScheduler_ListTable extends PP_List_Table {
+	/**
+	 * The package name.
+	 */
+	protected $package = 'action-scheduler';
+
+	/**
+	 * Columns to show (name => label).
+	 */
+	protected $columns = array();
+
+	/**
+	 * Actions (name => label).
+	 */
+	protected $row_actions = array();
+
+	/**
+	 *  The active data stores
+	 */
+	protected $store;
+
+	/**
+	 *  A logger to use for getting action logs to display
+	 */
+	protected $logger;
+
+	/**
+	 * Bulk actions. The key of the array is the method name of the implementation:
+	 *
+	 *     bulk_<key>(array $ids, string $sql_in).
+	 *
+	 * See the comments in the parent class for further details
+	 */
+	protected $bulk_actions = array();
+
+	/**
+	 * Flag variable to render our notifications, if any, once.
+	 */
+	protected static $did_notification = false;
+
+
+	/**
+	 * Sets the current data store object into `store->action` and initialises the object.
+	 */
+	public function __construct( $store, $logger ) {
+
+		$this->store  = $store;
+		$this->logger = $logger;
+
+		$this->maybe_render_admin_notices();
+
+		$this->bulk_actions = array(
+			'delete' => __( 'Delete', 'action-scheduler' ),
+		);
+
+		$this->columns = array(
+			'hook'   => __( 'Hook', 'action-scheduler' ),
+			'status' => __( 'Status', 'action-scheduler' ),
+			'args'   => __( 'Arguments', 'action-scheduler' ),
+			'group'  => __( 'Group', 'action-scheduler' ),
+			'recurrence' => __( 'Recurrence', 'action-scheduler' ),
+			'scheduled'  => __( 'Scheduled Date', 'action-scheduler' ),
+			'log'        => __( 'Log', 'action-scheduler' ),
+		);
+
+		$this->row_actions = array(
+			'hook' => array(
+				'run' => array(
+					__( 'Run', 'action-scheduler' ),
+					__( 'Process the action now as if it were run as part of a queue', 'action-scheduler' ),
+				),
+			),
+		);
+
+		parent::__construct( array(
+			'singular' => __( 'action-scheduler', 'action-scheduler' ),
+			'plural'   => __( 'action-scheduler', 'action-scheduler' ),
+			'ajax'     => false,
+		) );
+	}
+
+	/**
+	 * Convert a interval of seconds into a two part human friendly string.
+	 *
+	 * The WordPress human_time_diff() function only calculates the time difference to one degree, meaning
+	 * even if an action is 1 day and 11 hours away, it will display "1 day". This funciton goes one step
+	 * further to display two degrees of accuracy.
+	 *
+	 * Based on Crontrol::interval() funciton by Edward Dale: https://wordpress.org/plugins/wp-crontrol/
+	 *
+	 * @param int $interval A interval in seconds.
+	 * @return string A human friendly string representation of the interval.
+	 */
+	private static function human_interval( $interval ) {
+
+		// array of time period chunks
+		$chunks = array(
+			array( 60 * 60 * 24 * 365 , _n_noop( '%s year', '%s years', 'action-scheduler' ) ),
```

Comment:
Fixed both of those up with SHA: 32a68a12528e8ea9532ddd00db71ba9c2f89b74f and SHA: ce47a6d9a9f8ba826732350d104c485c0f18d1eb

I then rewrote `human_interval()` entirely with SHA: f7692983221e5aec5ff5e7c389a6568196613cb7 because it was gross, duplicating some code, and it seemed worth updating because it could be useful for more than just displaying the recurrence interval (so I used it for displaying the interval until a scheduled date too).